pydantic/_internal/_known_annotated_metadata.py

```diff
@@ -311,31 +314,19 @@ def collect_known_metadata(annotations: Iterable[Any]) -> tuple[dict[str, Any],
         #> ({'gt': 1, 'min_length': 42}, [Ellipsis])
         ```
     """
-    import annotated_types as at
-
     annotations = expand_grouped_metadata(annotations)
 
     res: dict[str, Any] = {}
     remaining: list[Any] = []
+
     for annotation in annotations:
         # isinstance(annotation, PydanticMetadata) also covers ._fields:_PydanticGeneralMetadata
         if isinstance(annotation, PydanticMetadata):
             res.update(annotation.__dict__)
         # we don't use dataclasses.asdict because that recursively calls asdict on the field values
-        elif isinstance(annotation, at.MinLen):
-            res.update({'min_length': annotation.min_length})
-        elif isinstance(annotation, at.MaxLen):
-            res.update({'max_length': annotation.max_length})
-        elif isinstance(annotation, at.Gt):
-            res.update({'gt': annotation.gt})
-        elif isinstance(annotation, at.Ge):
-            res.update({'ge': annotation.ge})
-        elif isinstance(annotation, at.Lt):
-            res.update({'lt': annotation.lt})
-        elif isinstance(annotation, at.Le):
-            res.update({'le': annotation.le})
-        elif isinstance(annotation, at.MultipleOf):
-            res.update({'multiple_of': annotation.multiple_of})
+        elif (annotation_type := type(annotation)) in (at_to_constraint_map := _get_at_to_constraint_map()):
+            constraint = at_to_constraint_map[annotation_type]
+            res[constraint] = getattr(annotation, constraint)
```

Comment:
And this is the other - lmk what you prefer @dmontagu

Reply:
Do you need to handle the possible `AttributeError` here? I saw a KeyError handled above, thought it might be related, but maybe not?

Reply:
Don't need to deal with the attribute error here bc we do the `in` check above. I'll go with this approach!